subprojects/core/src/test/groovy/org/gradle/execution/plan/DefaultExecutionPlanParallelTest.groovy

```diff
@@ -263,21 +264,25 @@ class DefaultExecutionPlanParallelTest extends AbstractExecutionPlanSpec {
         tasksAreNotExecutedInParallel(a, b)
     }
 
+    @ToBeImplemented("When we support symlinks in the VFS, we should implement this as well")
     @Requires(TestPrecondition.SYMLINKS)
     def "a task that writes into a symlink that overlaps with output of currently running task is not started"() {
         given:
         def taskOutput = file("outputDir").createDir()
         def symlink = file("symlink")
-        fs.createSymbolicLink(symlink, taskOutput)
+        symlink.createLink(taskOutput)
 
         and:
         Task a = task("a", type: AsyncWithOutputDirectory)
         _ * a.outputDirectory >> taskOutput
         Task b = task("b", type: AsyncWithOutputFile)
-        _ * b.outputFile >> symlink.file("fileUnderSymlink")
+        _ * b.outputFile >> new File(symlink, "fileUnderSymlink")
```

Comment:
Why is this better?

Reply:
`TestFile.file()` canonicalizes the resulting file. So the test actually passes when I use `symlink.file()`, since the location is not what one expects.

Reply:
I added a comment to that regard.